form/src/main/java/org/dhis2/form/data/FormRepositoryImpl.kt

```diff
@@ -22,7 +22,8 @@ class FormRepositoryImpl(
     private val dataEntryRepository: DataEntryRepository?,
     private val ruleEngineRepository: RuleEngineRepository?,
     private val rulesUtilsProvider: RulesUtilsProvider?,
-    private val legendValueProvider: LegendValueProvider?
+    private val legendValueProvider: LegendValueProvider?,
+    private val optionService: OptionSetDialogRepository
```

Comment:
I think that this is only used to send it to the `OptionSetDialogViewModel` via `OptionSetDialog`. Why don't inject directly in the `OptionSetDialogViewModel`?